CMake/SiloFindSzip.cmake

```diff
@@ -0,0 +1,91 @@
+# CONTRIBUTORS BE LIABLE FOR  ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR  CONSEQUENTIAL DAMAGES  (INCLUDING, BUT NOT  LIMITED TO,
+# PROCUREMENT OF  SUBSTITUTE GOODS  OR SERVICES; LOSS  OF USE,  DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+# LIABILITY, WHETHER  IN CONTRACT, STRICT LIABILITY,  OR TORT (INCLUDING
+# NEGLIGENCE OR  OTHERWISE) ARISING IN  ANY WAY OUT  OF THE USE  OF THIS
+# SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+# This work was produced at Lawrence Livermore National Laboratory under
+# Contract No.  DE-AC52-07NA27344 with the DOE.
+#
+# Neither the  United States Government nor  Lawrence Livermore National
+# Security, LLC nor any of  their employees, makes any warranty, express
+# or  implied,  or  assumes  any  liability or  responsibility  for  the
+# accuracy, completeness,  or usefulness of  any information, apparatus,
+# product, or  process disclosed, or  represents that its use  would not
+# infringe privately-owned rights.
+#
+# Any reference herein to  any specific commercial products, process, or
+# services by trade name,  trademark, manufacturer or otherwise does not
+# necessarily  constitute or imply  its endorsement,  recommendation, or
+# favoring  by  the  United  States  Government  or  Lawrence  Livermore
+# National Security,  LLC. The views  and opinions of  authors expressed
+# herein do not necessarily state  or reflect those of the United States
+# Government or Lawrence Livermore National Security, LLC, and shall not
+# be used for advertising or product endorsement purposes.
+#
+##############################################################################
+
+###
+# Uses find_package to find the szip library
+#   If SILO_SZIP_DIR is defined, uses it to tell CMake where to look.
+#   Use CONFIG version of find_package if szip-config.cmake exists.
+###
+
+
+if(DEFINED SILO_SZIP_DIR AND EXISTS ${SILO_SZIP_DIR}/szip-config.cmake)
+    # this works for szip with CMake
+    find_package(SZIP PATHS ${SILO_SZIP_DIR} CONFIG)
+else()
+    if(DEFINED SILO_SZIP_DIR AND EXISTS ${SILO_SZIP_DIR})
+        # help CMake find the specified szip
+        set(SZIP_ROOT ${SILO_SZIP_DIR})
+    endif()
+    find_package(SZIP)
+endif()
+
+if(SZIP_FOUND)
+    # needed for config.h
+    set(HAVE_LIBSZ 1)
+    if(WIN32 AND (SILO_ENABLE_SILEX OR SILEX_ENABLE_BROWSER))
+        get_target_property(SZIP_DLL szip IMPORTED_LOCATION_RELEASE )
+        install(FILES ${SZIP_DLL} DESTINATION bin
+                PERMISSIONS OWNER_READ OWNER_WRITE
+                            GROUP_READ GROUP_WRITE
+                            WORLD_READ)
+
+        add_custom_command(TARGET copy_deps POST_BUILD
+             COMMAND ${CMAKE_COMMAND} -E copy_if_different
+             ${SZIP_DLL} ${Silo_BINARY_DIR}/bin/$<CONFIG>)
+
+    endif()
+else()
+    message(FATAL_ERROR "Could not find szip, you may want to try setting SILO_SZIP_DIR")
```

Comment:
So, I am still a little fuzzy on why we need to *find* szip here.

As I understand it, we are using the same basic CMake find-this-package tooling to find an *indirect* dependency (one that comes from HDF5) so that we know to then link Silo with `-lsz` when we build executables, right?

But, this is a different beast than Qt, zlib and HDF5 because its an *indirect* dependency. Would you mind adjust relevant Silo related SZIP CMake variables to include HDF5 in the name? I mean `SILO_SZIP_DIR` becomes `SILO_HDF5_SZIP_DIR`. Does that make sense?

Reply:
In my limited testing of various builds of hdf5 on linux (pascal), you do not need to find szip in order for everything to build correctly.

However, when I build on Windows, we need may need to find szip.

The call to this find module in the root CMakeLists.txt:
```
##
# szip  if HDF5 was built with szip support on Windows (needed to find dll's)
##
if(WIN32 AND HDF5_ENABLE_SZIP_SUPPORT)
    include(SiloFindSzip)
endif()
```

Changing the name of the var is fine.

Unless you can think of a better way to handle this possible requirement on Windows that is more hidden from the general user.

Reply:
Oh yeah, you mentioned that last week  and I forgot about the larger context which triggers this logic. This is fine as is with the slight change in var name.

Reply:
Changed the name.